### PR TITLE
[faker]: Sync up arg types for system subobject

### PR DESCRIPTION
**Pull request by IvanGoncharov**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Marak/faker.js/blob/master/lib/system.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/faker/index.d.ts [hunk 1]
```diff
@@ -266,13 +266,13 @@ declare namespace Faker {
...
-            fileName(ext?: string, type?: string): string;
-            commonFileName(ext: string, type?: string): string;
+            fileName(): string;
+            commonFileName(ext?: string): string;
             mimeType(): string;
             commonFileType(): string;
             commonFileExt(): string;
             fileType(): string;
-            fileExt(mimeType: string): string;
+            fileExt(mimeType?: string): string;
```
Comment (IvanGoncharov) [inline types/faker/index.d.ts:275]: Made it optional, see here:
https://github.com/Marak/faker.js/blob/c9764abd20a198e318367952a94587eda1bac645/lib/system.js#L172

File: types/faker/index.d.ts [hunk 2]
```diff
@@ -266,13 +266,13 @@ declare namespace Faker {
         };
 
         system: {
-            fileName(ext?: string, type?: string): string;
-            commonFileName(ext: string, type?: string): string;
+            fileName(): string;
```
Comment (IvanGoncharov) [inline types/faker/index.d.ts:269]: See https://github.com/Marak/faker.js/blob/c9764abd20a198e318367952a94587eda1bac645/lib/system.js#L45

File: types/faker/index.d.ts [hunk 3]
```diff
@@ -266,13 +266,13 @@ declare namespace Faker {
...
+            commonFileName(ext?: string): string;
```
Comment (IvanGoncharov) [inline types/faker/index.d.ts:270]: Optional, see: https://github.com/Marak/faker.js/blob/c9764abd20a198e318367952a94587eda1bac645/lib/system.js#L64

File: types/faker/index.d.ts (hunk 2 above)
Comment (heavenshell) [reply types/faker/index.d.ts:269]: Please change tests.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/04657b253bcdd391b28709cbbf9bd8d78f81611c/types/faker/faker-tests.ts#L312

File: types/faker/index.d.ts (hunk 3 above)
Comment (heavenshell) [reply types/faker/index.d.ts:270]: ditto
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/04657b253bcdd391b28709cbbf9bd8d78f81611c/types/faker/faker-tests.ts#L313

File: types/faker/index.d.ts (hunk 1 above)
Comment (heavenshell) [reply types/faker/index.d.ts:275]: Please add new test.
```typescript
resultStr = faker.system.fileExt();

```

File: types/faker/index.d.ts (hunk 3 above)
Comment (heavenshell) [reply types/faker/index.d.ts:270]: Also please add optional test.
```typescript
resultStr = faker.system.commonFileName();
```

File: types/faker/index.d.ts (hunk 1 above)
Comment (heavenshell) [reply types/faker/index.d.ts:275]: Please add new optional test

```typescript
resultStr = faker.system.fileExt();
```

**Review by heavenshell — APPROVED** [review]
LGTM!

**Comment by heavenshell** [comment]
Ready to merge

**Pull request merged**